test/deploy.py

```diff
@@ -0,0 +1,569 @@
+            for volume in volumes:
+                devices[volume] = {
+                    "pool": pool_name,
+                    "source": volume,
+                    "type": "disk",
+                }
+            # Create profile.
+            print("Creating VM Profile {}".format(profile_name))
+            self.client.profiles.create(
+                profile_name, config["config"], config["devices"]
+            )
+            self.model["profile"] = profile_name
+
+    def create_vm(
+        self,
+        image="ubuntu/jammy",
+        flavor="c4-m10",
+        pool_name=deploy_tag,
+        profile_name=deploy_tag,
+        is_start=True,
+    ) -> string:
+        """Create a virtual machine for LXD."""
+        if not self.client:
+            raise AssertionError("LXD Client not available to runner.")
+
+        # Create Virtual Machine
+        vm_name = self.deploy_tag + "-" + _get_random_string(4)
+        config = {
+            "name": vm_name,
+            "instance_type": flavor,
+            "storage": pool_name,
+            "profiles": [profile_name],
+            "devices": {
+                "root": {
+                    "path": "/",
+                    "pool": pool_name,
+                    "size": "20GB",
+                    "type": "disk"
+                }
+            },
+            "source": {
+                "type": "image",
+                "certificate": "",
+                "alias": image,
+                "server": "https://images.linuxcontainers.org",
+                "protocol": "simplestreams",
+                "mode": "pull",
+                "allow_inconsistent": False,
+            },
+        }
+
+        # Create VM
+        print("Creating VM {}".format(vm_name))
+        self.client.virtual_machines.create(config, wait=True)
+        self.model["vm_name"] = vm_name
+
+        if is_start:
+            self.client.virtual_machines.get(vm_name).start(wait=True)
+
+        self.wait_for_vm_ready(vm_name)
+        # Increase Root Partition size on VM.
+        self.grow_root_partition(vm_name)
+        return vm_name  # vm_name to refer to the newly create vm.
+
+    def check_call_on_vm(
+        self, vm_name: string, cmd: list, is_fail_print=True
+    ) -> tuple:
+        """Execute Command on VM."""
+        if not self.client:
+            raise AssertionError("LXD Client not available to runner.")
+
+        if not self.client.instances.exists(vm_name):
+            raise AssertionError("VM {} does not exist.".format(vm_name))
+
+        inner_cmd = ["lxc", "exec", vm_name, "--", *cmd]
+        try:
+            subprocess.check_call(inner_cmd)
+        except subprocess.CalledProcessError as e:
+            if is_fail_print:
+                print("Failed Executing on {}: Output {}".format(vm_name, e))
+            raise e
+
+    def check_output_on_cephadm_shell(
+        self, vm_name: string, cmd: list, is_fail_print=True
+    ) -> dict:
```

Comment:
This function seems to return a `str`, not a `dict`.

Reply:
Thanks.